websites/A/Apple/presence.ts

```diff
@@ -1,145 +1,501 @@
-    document.title = document.getElementsByTagName("h1")[0][text];
-    if (document.title.startsWith("Buy")) {
-      const name = document.title.replace("Buy ", "");
-      presenceData.details = "Buying product";
-      presenceData.state = name;
-      presenceData.startTimestamp = time;
-      presence.setActivity(presenceData);
+  const urlpath = window.location.pathname.toLowerCase().split("/"),
+      setting = {
+        timeElapsed: await presence.getSetting("timeElapsed"),
+        buttons: await presence.getSetting("showButtons"),
+        logo: await presence.getSetting("logo"),
+        shopCheckout: await presence.getSetting("shopCheckout")
+      },
+      logoArr = ["logo", "logo-rainbow"],
+      langs = [
+        "bh",
+        "bh-ar",
+        "bw",
+        "cm",
+        "cf",
+        "ci",
+        "ch",
+        "ch-de",
+        "ch-fr",
+        "eg",
+        "eg-ar",
+        "gw",
+        "gn",
+        "gq",
+        "in",
+        "il",
+        "jo",
+        "jo-ar",
+        "ke",
+        "kw",
+        "kw-ar",
+        "mg",
+        "ml",
+        "ma",
+        "mu",
+        "mz",
+        "ne",
+        "ng",
+        "om",
+        "om-ar",
+        "qa",
+        "qa-ar",
+        "sa",
+        "sa-ar",
+        "sn",
+        "za",
+        "tn",
+        "ug",
+        "ae",
+        "ae-ar",
+        "au",
+        "hk/en",
+        "hk",
+        "id",
+        "jp",
+        "kr",
+        "mo",
+        "my",
+        "nz",
+        "ph",
+        "sg",
+        "tw",
+        "th",
+        "vn",
+        "am",
+        "az",
+        "by",
+        "benl",
+        "befr",
+        "bg",
+        "cz",
+        "dk",
+        "de",
+        "ee",
+        "es",
+        "fr",
+        "ge",
+        "gr",
+        "hr",
+        "ie",
+        "it",
+        "kz",
+        "kg",
+        "lv",
+        "li",
+        "lt",
+        "lu",
+        "hu",
+        "mt",
+        "md",
+        "me",
+        "nl",
+        "mk",
+        "no",
+        "at",
+        "pl",
+        "pt",
+        "ro",
+        "ru",
+        "sk",
+        "si",
+        "chde",
+        "chfr",
+        "fi",
+        "se",
+        "tj",
+        "tr",
+        "tm",
+        "uk",
+        "ua",
+        "uz",
+        "us",
+        "lae",
+        "lae",
+        "la",
+        "lae",
+        "lae",
+        "lae",
+        "la",
+        "br",
+        "lae",
+        "lae",
+        "cl",
+        "co",
+        "la",
+        "lae",
+        "la",
+        "la",
+        "la",
+        "lae",
+        "la",
+        "lae",
+        "la",
+        "lae",
+        "mx",
+        "lae",
+        "la",
+        "la",
+        "la",
+        "la",
+        "lae",
+        "lae",
+        "lae",
+        "lae",
+        "lae",
+        "lae",
+        "lae",
+        "la",
+        "la",
+        "la",
+        "lae",
+        "ca",
+        "lae",
+        "la"
+      ],
```

Comment:
well, you don't need this at all.

Reply:
Well, like I said in dms, you don't need it and I can prove it right away:
https://apple.com/mac
![image](https://user-images.githubusercontent.com/9348108/117290219-d39ecc80-ae6d-11eb-8ba4-9561dc432603.png)
https://apple.com/pl/buy-mac/imac
![image](https://user-images.githubusercontent.com/9348108/117290506-2bd5ce80-ae6e-11eb-88cc-967cf36521a4.png)
That's shows that you don't need array since you can just check if it includes, and it's not really problematic, or is it?
Old code:
```javascript
const urlpNum = langs.indexOf(urlpath[1]) > -1 ? 2 : 1
 if(urlpath[urlpNum] !== 'shop' ...)
```
new:
```javascript
if(urlpath.includes("shop")
```
is it really hard to understand? You only bloat your code.
Next example - home page:
Old code:
```javascript
if(!urlpath[urlpNum])
```
new:
```javascript
if(urlpath === "/")
```
Comparing & viewing product
old code:
```javascript
else if(products.indexOf(urlpath[urlpNum]) > -1) {
      if(urlpath[urlpNum + 1] === 'compare') {
```

new:
```javascript
products.find(e => urlpath.includes(e))
if(urlpath.includes("compare"))
```
full example:
![image](https://user-images.githubusercontent.com/9348108/117291782-b7039400-ae6f-11eb-872c-d490f146c000.png)

And that's my last comment about it. You can leave `products` and `services` array but langs array is unnecessary like I proved above.